### PR TITLE
Ltd 3431 remove additional actions

### PR DESCRIPTION
**Pull request by kevincarrogan**
### Aim

This hides additional actions for a case worker that isn't allocated to a case.

[LTD-3431](https://uktrade.atlassian.net/browse/LTD-3431)

[LTD-3431]: https://uktrade.atlassian.net/browse/LTD-3431?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by currycoder — COMMENTED** [review]
Just a few comments/questions throughout to help my understanding.

One other thought I had while looking through this is, could it be helpful to share this early with UCD?  e.g. by popping it on demo.  Previously, we have had back/forth on how permissions should work and I'm wondering if this might expediate the process overall to get some early feedback before this merges in/while things are easy to adjust.

**Review by antroy-madetech — APPROVED** [review]
LGTM. Just a question for the benefit of my understanding.

File: unit_tests/caseworker/advice/views/test_advice_view.py [hunk 1]
```diff
@@ -79,4 +105,30 @@ def test_advice_view_shows_assessed_trigger_list_goods_if_all_are_assessed(
...
+    case = data_standard_case_with_all_trigger_list_products_assessed
+    mock_application(case)
+
+    response = authorized_client.get(url)
+
+    assert response.status_code == 200
+
+    soup = BeautifulSoup(response.content, "html.parser")
+    product_table = soup.find(id="assessed-products")
+    assert product_table is None
```
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply unit_tests/caseworker/advice/views/test_advice_view.py:131]: This isn’t the TAU assessment, it’s the new BEIS assessment that this is hidden.

File: caseworker/tau/templates/tau/edit.html [hunk 2]
```diff
@@ -21,6 +23,10 @@ <h3 class="govuk-heading-m">{{ good.line_number }}. {{ good.good.name }}</h3>
                     {% csrf_token %}
                     <h1 class="govuk-heading-m tau__headline">Assessing {{good.good.name}}</h1>
                     {% crispy form %}
+                    {% test_rule 'can_user_assess_products' current_user case as can_user_assess_products %}
+                    {% if can_user_assess_products %}
+                        <input type="submit" name="submit" value="Save" class="govuk-button" id="submit-id-submit" />
```
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply caseworker/tau/templates/tau/edit.html:28]: Correct.

File: caseworker/core/rules.py [hunk 3]
```diff
@@ -17,5 +17,15 @@ def is_user_assigned(user, case):
...
+rules.add_rule("can_user_change_case", is_user_allocated)
+rules.add_rule("can_user_move_case_forward", is_user_allocated)
+rules.add_rule("can_user_review_and_countersign", is_user_allocated)
+rules.add_rule("can_user_review_and_combine", is_user_allocated)
+rules.add_rule("can_user_make_recommendation", is_user_allocated)
+rules.add_rule("can_user_assess_products", is_user_allocated)
+rules.add_rule("can_user_add_an_ejcu_query", is_user_allocated)
+rules.add_rule("can_user_attach_document", is_user_allocated)
+rules.add_rule("can_user_generate_document", is_user_allocated)
+rules.add_rule("can_user_add_contact", is_user_allocated)
```
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply caseworker/core/rules.py:31]: That is correct. 

More granular rules mean we have the ability to toggle thing independently in the future. At the moment they all work off the same predicates but if we want to make any changes in the future to just one of these rules then we just need to tweak the predicate for that rule and that rule alone.

File: ui_tests/caseworker/features/give_advice.feature [hunk 4]
```diff
@@ -227,6 +227,9 @@ Feature: I want to record my user advice and any comments and conditions relatin
     And I change my team to "Licensing Unit" and default queue to "Licensing Unit Post-circulation Cases to Finalise"
     And I go to my case list
     And I click the application previously created
+    And I assign myself to the case
+    And I go to my case list
+    And I click the application previously created
```
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply ui_tests/caseworker/features/give_advice.feature:232]: Yeah. Originally I just had the assign step here but then the tests weren’t properly picking up from where they needed to. This just gets us back on the right track. This is what the user would kind of have to do at the moment as well.

**Comment by kevincarrogan** [comment]
> Just a few comments/questions throughout to help my understanding.
> 
> One other thought I had while looking through this is, could it be helpful to share this early with UCD? e.g. by popping it on demo. Previously, we have had back/forth on how permissions should work and I'm wondering if this might expediate the process overall to get some early feedback before this merges in/while things are easy to adjust.

I can certainly do that.

File: unit_tests/caseworker/advice/views/test_advice_view.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (currycoder) [reply unit_tests/caseworker/advice/views/test_advice_view.py:131]: Ahhh thanks - makes sense now

**Review by currycoder — APPROVED** [review]
Thanks for responding; approved pending popping this on demo for UCD's input

**Pull request merged**